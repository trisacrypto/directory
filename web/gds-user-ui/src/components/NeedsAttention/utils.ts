export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
