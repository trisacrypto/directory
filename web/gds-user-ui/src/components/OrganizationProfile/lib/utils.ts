export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
