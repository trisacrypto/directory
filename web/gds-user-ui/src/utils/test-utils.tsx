import { ChakraProvider } from '@chakra-ui/react';
import { render as rtlRender, RenderOptions as RTLRenderOptions } from '@testing-library/react';
import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { Provider } from 'react-redux';
import customTheme from './theme';
import rootReducer from 'application/store/rootReducer';
import { configureStore, Store } from '@reduxjs/toolkit';

type RenderOptions = {
  rhfDefaultValues?: { [x: string]: any };
  store?: Store;
  preloadedState?: StoreState;
  renderOptions?: Omit<RTLRenderOptions, 'wrapper'>;
};

type StoreState = {
  [x: string]: any;
};

export function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {
    rhfDefaultValues,
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: RenderOptions = {}
) {
  function Wrapper({ children }: { children: React.ReactNode }) {
    const methods = useForm({ defaultValues: rhfDefaultValues });
    return (
      <Provider store={store}>
        <ChakraProvider theme={customTheme}>
          <FormProvider {...methods}>{children}</FormProvider>
        </ChakraProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
