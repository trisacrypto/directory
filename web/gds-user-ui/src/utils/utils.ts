import { left } from '@popperjs/core';
import _ from 'lodash';
import { RegistrationAuthority, StepStatus } from 'types/type';
import { TStep } from './localStorageHelper';
import registrationAuthority from './registration-authority.json';

const DEFAULT_REGISTRATION_AUTHORITY = 'RA777777';

export const findStepKey = (steps: any, key: number) =>
  steps?.filter((step: any) => step.key === key);

export const isValidUuid = (str: string) => {
  // Regular expression to check if string is a valid UUID
  const regexExp =
    /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi;
  return regexExp.test(str);
};

export const getStepData = (steps: any, key: number): TStep | undefined => {
  const s = findStepKey(steps, key);
  if (s && s?.length === 1) {
    return s[0].data;
  }
  return undefined;
};

export const getStepStatus = (steps: any, key: number): StepStatus | undefined => {
  const s = findStepKey(steps, key);
  if (s && s?.length === 1) {
    return s[0].status;
  }
  return undefined;
};

export const hasStepError = (steps: any): boolean => {
  const s = steps.filter((step: any) => step.status === 'error');
  return s.length > 0;
};

export const getStepDatas = (steps: any) => {
  const s = steps
    ?.map((step: any) => step.data)
    .reduce((acc: any, cur: any) => ({ ...acc, ...cur }), {});

  return { ...s };
};

export const getValueByPathname = (obj: Record<string, any>, path: string) => {
  return _.get(obj, path);
};

export const getDomain = (url: string | URL) => {
  try {
    const _url = new URL(url);
    return _url?.hostname?.replace('www.', '');
  } catch (error) {
    console.error('[error]', error);
    return null;
  }
};

export const getRegistrationAuthorities = () => [...new Set(registrationAuthority)];

export const getRegistrationAuthoritiesOptions = (country?: any) => {
  const newArray = [...Array.from(new Set(registrationAuthority))];
  if (country) {
    return newArray
      .filter(
        (v: RegistrationAuthority) =>
          v.country === country || v.option === DEFAULT_REGISTRATION_AUTHORITY
      )
      .map((v: RegistrationAuthority) => {
        const label = v.organization ? `${v.option} - ${v.organization}` : `${v.option}`;
        const l =
          v.jurisdiction && v.jurisdiction !== v.country_name
            ? `${label} - ${v.jurisdiction}`
            : label;
        return {
          value: v.option,
          label: l,
          isDisabled: !!v.isDisabled
        };
      });
  }
  return newArray.map((v: RegistrationAuthority) => {
    const label = v.organization ? `${v.option} - ${v.organization}` : `${v.option}`;
    return {
      value: v.option,
      label,
      isDisabled: !!v.isDisabled
    };
  });
};

export const mapTrixoFormForBff = (data: any) => {
  const { trixo } = data;
  const { applicable_regulations, other_jurisdictions } = trixo;

  const cleanAppRegulation = applicable_regulations.reduce((acc: any, value: any) => {
    if (value.name.length > 0) {
      acc.push(value.name);
    }
    return acc;
  }, []);
  const cleanOtherJurisdiction = other_jurisdictions.filter(
    (o: any) => o?.country?.length > 0 && o?.regulator_name?.length > 0
  );

  return {
    ...data,
    trixo: {
      ...trixo,
      applicable_regulations: cleanAppRegulation.length > 0 ? cleanAppRegulation : [],
      other_jurisdictions: cleanOtherJurisdiction.length > 0 ? cleanOtherJurisdiction : []
    }
  };
};

export const hasValue = (obj: Record<string, any>) => {
  return obj && Object.values(obj).some(Boolean);
};

export const getColorScheme = (status: string) => {
  if (status === 'yes' || status) {
    return 'cyan';
  } else {
    return '#eee';
  }
};

export function currencyFormatter(
  amount: number | bigint,
  { style = 'currency', currency = 'USD' }
) {
  const formatedAmount = new Intl.NumberFormat('en-US', {
    style,
    currency
  });

  return formatedAmount.format(amount);
}
