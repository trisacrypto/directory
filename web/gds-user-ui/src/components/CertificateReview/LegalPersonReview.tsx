import React, { FC, useEffect } from 'react';
import {
  Stack,
  Box,
  Text,
  Heading,
  Table,
  Tbody,
  Tr,
  Td,
  Button,
  Tag,
  Divider
} from '@chakra-ui/react';
import { colors } from 'utils/theme';
import { useDispatch, useSelector, RootStateOrAny } from 'react-redux';
import { getStepData } from 'utils/utils';
import { loadDefaultValueFromLocalStorage, TStep } from 'utils/localStorageHelper';
import useCertificateStepper from 'hooks/useCertificateStepper';
import { getNameIdentiferTypeLabel } from 'constants/name-identifiers';
import { getNationalIdentificationLabel } from 'constants/national-identification';
import { COUNTRIES } from 'constants/countries';
interface LegalReviewProps {}
// NOTE: need some clean up.
const isValidIvmsAddress = (address: any) => {
  if (address) {
    return !!(address.country && address.address_type);
  }
  return false;
};

const hasAddressLine = (address: any) => {
  if (isValidIvmsAddress(address)) {
    return Array.isArray(address.address_line) && address.address_line.length > 0;
  }
  return false;
};

const hasAddressField = (address: any) => {
  if (isValidIvmsAddress(address) && !hasAddressLine(address)) {
    return !!(address.street_name && (address.building_number || address.building_name));
  }
  return false;
};

const hasAddressFieldAndLine = (address: any) => {
  if (hasAddressField(address) && hasAddressLine(address)) {
    console.warn('[ERROR]', 'cannot render address');
    return true;
  }
  return false;
};

export const renderLines = (address: any) => (
  <address data-testid="addressLine">
    {address.address_line.map(
      (addressLine: any, index: number) => addressLine && <div key={index}>{addressLine} </div>
    )}
    <div>{address?.country}</div>
  </address>
);

export const renderField = (address: any) => (
  <address data-testid="addressField">
    {address.sub_department ? (
      <>
        {address?.sub_department} <br />
      </>
    ) : null}
    {address.department ? (
      <>
        {address?.department} <br />
      </>
    ) : null}
    {address.building_number} {address?.street_name}
    <br />
    {address.post_box ? (
      <>
        P.O. Box: {address?.post_box} <br />
      </>
    ) : null}
    {address.floor || address.room || address.building_name ? (
      <>
        {address?.floor} {address?.room} {address?.building_name} <br />
      </>
    ) : null}
    {address.district_name ? (
      <>
        {address?.district_name} <br />
      </>
    ) : null}
    {address.town_name || address.town_location_name || address.country_sub_division ? (
      <>
        {address?.town_name} {address?.town_location_name} {address?.country_sub_division}{' '}
        {address?.post_code} <br />
      </>
    ) : null}
    {address?.country}
  </address>
);

const renderAddress = (address: any) => {
  if (hasAddressFieldAndLine(address)) {
    console.warn('[ERROR]', 'invalid address with both fields and lines');
    return <div>Invalid Address</div>;
  }

  if (hasAddressLine(address)) {
    return renderLines(address);
  }

  if (hasAddressField(address)) {
    return renderField(address);
  }

  console.warn('[ERROR]', 'could not render address');
  return <div>Unparseable Address</div>;
};

const LegalPersonReview: React.FC<LegalReviewProps> = (props) => {
  const { jumpToStep } = useCertificateStepper();
  const steps: TStep[] = useSelector((state: RootStateOrAny) => state.stepper.steps);
  const [legalPerson, setLegalPerson] = React.useState<any>({});

  useEffect(() => {
    const getStepperData = loadDefaultValueFromLocalStorage();
    const stepData = {
      ...getStepperData.entity
    };
    setLegalPerson(stepData);
  }, [steps]);
  return (
    <Box
      border="1px solid #DFE0EB"
      fontFamily={'Open Sans'}
      color={'#252733'}
      bg={'white'}
      fontSize={18}
      p={5}
      px={5}>
      <Stack>
        <Box display={'flex'} justifyContent="space-between" pt={4} ml={5}>
          <Heading fontSize={20}>Section 2: Legal Person</Heading>
          <Button
            bg={colors.system.blue}
            color={'white'}
            height={'34px'}
            onClick={() => jumpToStep(2)}
            _hover={{
              bg: '#10aaed'
            }}>
            {' '}
            Edit{' '}
          </Button>
        </Box>
        <Stack fontSize={18}>
          <Table
            sx={{
              'td:nth-child(2),td:nth-child(3)': { fontWeight: 'bold', paddingLeft: 0 },
              Tr: { borderStyle: 'hidden' }
            }}>
            <Tbody
              sx={{
                '*': {
                  fontSize: '1rem'
                }
              }}>
              <Tr>
                <Td fontSize={'1rem'} fontWeight="bold" colSpan={3}>
                  <Text mb={1}>Name Identifiers</Text>
                  <Divider />
                </Td>
              </Tr>
              <Tr>
                <Td fontStyle={'italic'} pt={0}>
                  The name and type of name by which the legal person is known.
                </Td>
                <Td>
                  <Tr>
                    {legalPerson.name?.name_identifiers?.map(
                      (nameIdentifier: any, index: number) => {
                        return (
                          <React.Fragment key={index}>
                            <Td paddingLeft={0} border="none">
                              {nameIdentifier.legal_person_name || 'N/A'}
                            </Td>
                            <Td paddingLeft={0} border="none">
                              {getNameIdentiferTypeLabel(
                                nameIdentifier.legal_person_name_identifier_type
                              )}
                            </Td>
                          </React.Fragment>
                        );
                      }
                    )}
                  </Tr>
                  <>
                    {legalPerson.name?.local_name_identifiers?.map(
                      (nameIdentifier: any, index: number) => {
                        return (
                          <React.Fragment key={index}>
                            <Td paddingLeft={0} pt={0} border="none">
                              {nameIdentifier.legal_person_name}
                            </Td>
                            <Td paddingLeft={0} pt={0} border="none">
                              {getNameIdentiferTypeLabel(
                                nameIdentifier.legal_person_name_identifier_type
                              )}
                            </Td>
                          </React.Fragment>
                        );
                      }
                    )}
                  </>
                  <>
                    {legalPerson.name?.phonetic_name_identifiers?.map(
                      (nameIdentifier: any, index: number) => {
                        return (
                          <React.Fragment key={index}>
                            <Td paddingLeft={0} pt={0} border="none">
                              {nameIdentifier.legal_person_name}
                            </Td>
                            <Td paddingLeft={0} pt={0} border="none">
                              {getNameIdentiferTypeLabel(
                                nameIdentifier.legal_person_name_identifier_type
                              )}
                            </Td>
                          </React.Fragment>
                        );
                      }
                    )}
                  </>
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Addresses</Td>
                <Td>
                  <Tr>
                    <Td paddingLeft={0} pt={0}>
                      {legalPerson?.geographic_addresses?.map((address: any, index: number) => (
                        <React.Fragment key={index}>{renderAddress(address)}</React.Fragment>
                      ))}
                      {/* {legalPerson?.geographic_addresses?.[0]?.address_line.map(
                        (line: any, i: any) => {
                          return <Text key={i}>{line}</Text>;
                        }
                      )} */}
                    </Td>
                    <Td pt={0}>{legalPerson?.geographic_addresses?.[0] && 'Legal Person'}</Td>
                  </Tr>
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Country of Registration</Td>
                <Td paddingLeft={0} pt={0}>
                  {(COUNTRIES as any)[legalPerson?.country_of_registration] || 'N/A'}
                </Td>
              </Tr>
              <Tr>
                <Td fontSize={'1rem'} fontWeight="bold" colSpan={3}>
                  <Text mb={1}>National Identification</Text>
                  <Divider />
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Identification Number</Td>
                <Td paddingLeft={0}>{legalPerson?.national_identification?.national_identifier}</Td>
              </Tr>
              <Tr>
                <Td pt={0}>Identification Type</Td>
                <Td pt={0}>
                  <Tag color={'white'} bg={'blue.400'} size={'lg'}>
                    {getNationalIdentificationLabel(
                      legalPerson?.national_identification?.national_identifier_type
                    )}
                  </Tag>
                </Td>
              </Tr>
              <Tr>
                <Td>Country of Issue</Td>
                <Td>
                  {(COUNTRIES as any)[legalPerson?.national_identification?.country_of_issue] ||
                    'N/A'}
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Reg Authority</Td>
                <Td pt={0}>
                  {legalPerson?.national_identification?.registration_authority || 'N/A'}
                </Td>
              </Tr>
            </Tbody>
          </Table>
        </Stack>
      </Stack>
    </Box>
  );
};
LegalPersonReview.defaultProps = {
  data: {}
};
export default LegalPersonReview;
