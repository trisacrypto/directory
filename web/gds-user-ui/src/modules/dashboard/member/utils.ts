import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

