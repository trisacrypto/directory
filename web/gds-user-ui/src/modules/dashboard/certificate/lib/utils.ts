import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
