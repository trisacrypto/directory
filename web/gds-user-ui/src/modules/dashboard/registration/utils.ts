import { getDefaultValue } from 'components/BasicDetailsForm/validation';
import { getRegistrationDefaultValues } from 'modules/dashboard/certificate/lib';
import {
  postRegistrationData,
  getRegistrationData,
  getSubmissionStatus
} from 'modules/dashboard/registration/service';
import { handleError } from 'utils/utils';

export const postRegistrationValue = (data: any) => {
  console.log('[postRegistrationValue]', data);
  return new Promise((resolve, reject) => {
    postRegistrationData(data)
      .then((res) => {
        console.log('[postRegistrationData]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        console.log('[postRegistrationData]', err);
        reject(err);
      });
  });
};
export const getRegistrationDefaultValue = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return regData.data;
    }
    const defaultValue: any = localStorage.getItem('certificateForm');
    if (defaultValue) {
      const val = JSON.parse(defaultValue);
      const postData = await postRegistrationData(val);
      if (postData.status === 204) {
        const getData = await getRegistrationData();
        localStorage.removeItem('certificateForm');
        return getData.data;
      }
    }
    const v = getRegistrationDefaultValues();
    await postRegistrationValue(v);
    return getRegistrationDefaultValues();
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

export const setRegistrationDefaultValue = () => {
  console.log('[setRegistrationDefaultValue]');
  const defaultValue: any = getRegistrationDefaultValues();
  return new Promise((resolve, reject) => {
    postRegistrationData(defaultValue)
      .then((res) => {
        // console.log('[default postRegistration value]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        handleError(err, 'failed to post registration value');
        reject(err);
      });
  });
};

// get registration data from backend and download the file
export const downloadRegistrationData = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      const blob = new Blob([JSON.stringify(regData.data)], { type: 'application/json' });
      const url = window.URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.setAttribute('download', 'registration.json');
      document.body.appendChild(link);
      link.click();
    }
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

// load default stepper

export const getDefaultStepper = async () => {
  try {
    const [regData, regStatus] = await Promise.all([getRegistrationData(), getSubmissionStatus()]);
    console.log('[regStatus]', regStatus.data);
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return {
        currentStep: regData.data.state.current,
        steps: regData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: regData.data.state.ready_to_submit,
        testnetSubmitted: regStatus?.data?.testnetSubmitted || false,
        mainnetSubmitted: regStatus?.data?.mainnetSubmitted || false
      };
    }
    const defaultValue: any = {
      current: 1,
      steps: [
        {
          key: 1,
          status: 'progress'
        }
      ]
    };
    // update registrations state object
    const postData = await postRegistrationData({ state: defaultValue });
    if (postData.status === 204) {
      const getData = await getRegistrationData();
      return {
        currentStep: getData.data.state.current,
        steps: getData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: false,
        testnetSubmitted: false,
        mainnetSubmitted: false
      };
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};

// // load default stepper without async call
// export const loadDefaultStepperSync = () => {
//   return Promise.resolve({
//     currentStep: 1,
//     steps: [
//       {
//         key: 1,
//         status: 'progress'
//       }
//     ],
//     lastStep: null,
//     hasReachSubmitStep: false
//   });
// };

// set stepper data
export const getRegistrationAndStepperData = async () => {
  try {
    const [regData, regStatus] = await Promise.all([
      getRegistrationDefaultValue(),
      getSubmissionStatus()
    ]);
    if (regData) {
      const response: any = {
        registrationData: regData,
        stepperData: {
          currentStep: regData?.state?.current || 1,
          steps: regData?.state?.steps || [{ key: 1, status: 'progress' }],
          lastStep: null,
          hasReachSubmitStep: regData?.state?.ready_to_submit || false,
          testnetSubmitted: !!regStatus?.data?.testnet_submitted,
          mainnetSubmitted: !!regStatus?.data?.mainnet_submitted
        }
      };
      return response;
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};
