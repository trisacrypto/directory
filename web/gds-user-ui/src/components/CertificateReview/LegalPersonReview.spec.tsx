import { render } from 'utils/test-utils';
import LegalPersonReview from './LegalPersonReview';

const defaultValues = {
  entity: {
    country_of_registration: 'AF',
    name: {
      name_identifiers: [
        {
          legal_person_name: 'KYC',
          legal_person_name_identifier_type: 'LEGAL_PERSON_NAME_TYPE_CODE_LEGL'
        },
        { legal_person_name: '', legal_person_name_identifier_type: '' }
      ],
      local_name_identifiers: [],
      phonetic_name_identifiers: []
    },
    geographic_addresses: [
      {
        address_type: 'ADDRESS_TYPE_CODE_BIZZ',
        address_line: ['a', 'Address 2', 'Address 3'],
        country: 'AO'
      },
      {
        address_type: 'ADDRESS_TYPE_CODE_HOME',
        address_line: ['705 Ryan Street', 'Sylvania', 'OH 43560'],
        country: 'US'
      },
      {
        address_type: 'ADDRESS_TYPE_CODE_HOME',
        address_line: ['11 Garfield St.', 'Libertyville', 'IL 60048'],
        country: 'US'
      }
    ],
    national_identification: {
      national_identifier_type: 'NATIONAL_IDENTIFIER_TYPE_CODE_TXID',
      country_of_issue: '',
      registration_authority: 'RA777777',
      national_identifier: '2'
    }
  },
  website: 'http://kyc.com',
  business_category: 'GOVERNMENT_ENTITY',
  vasp_categories: ['P2P', 'Kiosk'],
  established_on: '2022-04-22',
  organization_name: 'KYC'
};

describe('<LegalPersonReview />', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('should match snapshot', () => {
    localStorage.setItem('certificateForm', JSON.stringify(defaultValues));

    const { container } = render(<LegalPersonReview />);

    expect(localStorage.getItem).toHaveBeenCalledWith('certificateForm');
    expect(Object.keys(localStorage.__STORE__).length).toBe(1);

    expect(container).toMatchSnapshot();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
