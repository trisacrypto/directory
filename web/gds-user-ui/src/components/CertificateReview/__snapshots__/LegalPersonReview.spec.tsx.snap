// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegalPersonReview /> should match snapshot 1`] = `
<div>
  <div
    class="css-18e1pym"
  >
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="css-18uk989"
      >
        <h2
          class="chakra-heading css-13m0iy4"
        >
          Section 2: Legal Person
        </h2>
        <button
          class="chakra-button css-qhu6kf"
          type="button"
        >
          Edit
        </button>
      </div>
      <div
        class="chakra-stack css-16ztyi4"
      >
        <table
          class="chakra-table css-2zf4qf"
          role="table"
        >
          <tbody
            class="css-1sxjzw4"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1b8pwgj"
                colspan="3"
                role="gridcell"
              >
                Name Identifiers
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-oabm8g"
                role="gridcell"
              >
                The name and type of name by which the legal person is known.
              </td>
              <td
                class="css-xumdn4"
                role="gridcell"
              >
                <tbody
                  class="css-0"
                >
                  <tr
                    class="css-0"
                    role="row"
                  >
                    <td
                      class="css-1pkyvl"
                      role="gridcell"
                    >
                      KYC
                    </td>
                    <td
                      class="css-uluo24"
                      role="gridcell"
                    >
                      (
                      Legal
                      )
                    </td>
                    <td
                      class="css-1pkyvl"
                      role="gridcell"
                    >
                      N/A
                    </td>
                  </tr>
                </tbody>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1b1t9fp"
                role="gridcell"
              >
                Addresses
              </td>
              <td
                class="css-i257lr"
                role="gridcell"
              >
                <tr
                  class="css-0"
                  role="row"
                >
                  <td
                    class="css-1g75rv3"
                    role="gridcell"
                  >
                    <p
                      class="chakra-text css-187crao"
                    >
                       
                      Address 
                      1
                       : 
                      Business
                    </p>
                    <div
                      data-testid="addressLine"
                    >
                      <div>
                        a
                         
                      </div>
                      <div>
                        Address 2
                         
                      </div>
                      <div>
                        Address 3
                         
                      </div>
                      <div>
                        AO
                      </div>
                    </div>
                    <p
                      class="chakra-text css-187crao"
                    >
                       
                      Address 
                      2
                       : 
                      Residential
                    </p>
                    <div
                      data-testid="addressLine"
                    >
                      <div>
                        705 Ryan Street
                         
                      </div>
                      <div>
                        Sylvania
                         
                      </div>
                      <div>
                        OH 43560
                         
                      </div>
                      <div>
                        US
                      </div>
                    </div>
                    <p
                      class="chakra-text css-187crao"
                    >
                       
                      Address 
                      3
                       : 
                      Residential
                    </p>
                    <div
                      data-testid="addressLine"
                    >
                      <div>
                        11 Garfield St.
                         
                      </div>
                      <div>
                        Libertyville
                         
                      </div>
                      <div>
                        IL 60048
                         
                      </div>
                      <div>
                        US
                      </div>
                    </div>
                  </td>
                </tr>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-1b1t9fp"
                role="gridcell"
              >
                Country of Registration
              </td>
              <td
                class="css-18jx7ha"
                role="gridcell"
              >
                <tbody
                  class="css-0"
                >
                  <tr
                    class="css-0"
                    role="row"
                  >
                    <td
                      class="css-1uy98i6"
                      role="gridcell"
                    >
                      Afghanistan
                    </td>
                  </tr>
                </tbody>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-xumdn4"
                role="gridcell"
              />
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-a3eb33"
                colspan="3"
                role="gridcell"
              >
                <p
                  class="chakra-text css-92ucc9"
                >
                  National Identification
                </p>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-xumdn4"
                role="gridcell"
              />
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-oabm8g"
                role="gridcell"
              >
                Identification Number
              </td>
              <td
                class="css-i257lr"
                role="gridcell"
              >
                2
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-oabm8g"
                role="gridcell"
              >
                Identification Type
              </td>
              <td
                class="css-oabm8g"
                role="gridcell"
              >
                <span
                  class="css-vnm0v1"
                >
                  Tax Identification Number
                </span>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-xumdn4"
                role="gridcell"
              >
                Country of Registration
              </td>
              <td
                class="css-xumdn4"
                role="gridcell"
              >
                Afghanistan
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-oabm8g"
                role="gridcell"
              >
                Reg Authority
              </td>
              <td
                class="css-oabm8g"
                role="gridcell"
              >
                RA777777
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
